modules/common/src/main/scala/notebook/front/widgets/package.scala

```diff
@@ -307,85 +322,114 @@ package object widgets {
     }
   }
 
-  abstract class Chart[C:ToPoints:Sampler] extends JsWorld[Seq[(String, Any)], Seq[(String, Any)]] {
-    import notebook.JSBus._
-
-    def originalData:C
-    def maxPoints:Int
-
+  abstract class OriginalDataToSeqConversions[C: ToPoints : Sampler](originalData: C, maxPoints: Int)
```

Comment:
I was also thinking about something even more specific like `OriginalDataToRenderableConverter`.
i.e. `type Renderable = Seq[MagicRenderPoint]` (not sure if the type alias would help, maybe not).

what do you think?

actually it's still too much coupled with with `JsWorld/Widget`, but wouldn't be easy to refactor...